test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageBuilderTest.cs

```diff
@@ -1318,6 +1317,96 @@ public void SavingPackageValidatesReferences()
                 "Invalid assembly reference 'Bar.dll'. Ensure that a file named 'Bar.dll' exists in the lib directory.");
         }
 
+        [Fact]
+        public void SavingPackageValidatesMissingTPVInReferences()
+        {
+            // Arrange
+            var builder = new PackageBuilder
+            {
+                Id = "A",
+                Version = NuGetVersion.Parse("1.0"),
+                Description = "Test",
+            };
+            builder.Authors.Add("Test");
+            builder.Files.Add(new PhysicalPackageFile { TargetPath = @"lib\net5.0-windows\Foo.dll" });
+            builder.PackageAssemblyReferences = new[] { new PackageReferenceSet(NuGetFramework.Parse("net5.0-windows"), new string[] { "Foo.dll" }) };
+
+            ExceptionAssert.Throws<PackagingException>(() => builder.Save(new MemoryStream()),
+                "Some reference group TFMs are missing a platform version: net5.0-windows");
+        }
+
+        [Fact]
+        public void SavingPackageValidatesMissingTPVInFiles()
+        {
+            // Arrange
+            var builder = new PackageBuilder
+            {
+                Id = "A",
+                Version = NuGetVersion.Parse("1.0"),
+                Description = "Test",
+            };
+            builder.Authors.Add("Test");
+            builder.Files.Add(new PhysicalPackageFile { TargetPath = @"lib\net5.0-windows\Foo.dll" });
```

Comment:
These tests look great, but I'd like to encourage you to make them a bit more complete, in particular add tests for every potential file path we're validating (ie, lib, ref, runtimes etc).